### ISSUE TITLE
Consistent handling of Tx fees/minimum amounts

### ISSUE THREAD

**Issue by gomesalexandre** [shapeshift/web#1966]
### Overview

Currently, handling Tx fees and minimum amounts across the app is either inconsistent, or not implemented at all.

We should implement a consistent way to handle:

- minimum amounts to send/stake
  - depending on to the asset precision
  - depending on the account's balance
- max possible amounts to send/stake, including deducted gas fees

This way, we will be able to have guarantees in validating send/staking forms and never product an invalid Tx/ (from e.g percentages) or allow to continue with user input that would result in a failing Tx.

### References and additional details

This is taking over #1703 in a systematic, specced way vs. just tackling it for Cosmos.

### Acceptance Criteria

- Minimum amounts are handled across all send and staking modals
- Max sends are working across all send and staking modals, and do not produce errors on max sends because of the lack of network fees subtraction 
- When the resulting amount is effectively negative, it is zero'd out

**Comment by gomesalexandre** [comment]
Blocked by DeFi UI PR https://github.com/shapeshift/web/issues/2219
We could start work on this after the aforementioned PR is merged

**Comment by willyogo** [comment]
With #2219 merged, I'm removing the blocked and low priority tags and recommend prioritizing this. Cosmos staking is one of our strongest rev streams, and often users want to stake the Max amount, but currently this happy path results in an error.

**Comment by gomesalexandre** [comment]
@willyogo The max. ATOM issue has been fixed on the lib side, meaning users can now send max. amount of ATOM with the conservative fee properly deducted

<img width="464" alt="image" src="https://user-images.githubusercontent.com/17035424/193572855-7a0fb48e-0ac8-4211-ad68-86aef0549ee6.png">
<img width="401" alt="image" src="https://user-images.githubusercontent.com/17035424/193572881-30e34c4e-a525-40b3-b35d-d578d68a2b2f.png">

IMO, with this now fixed, we should be able to close this issue.
The issue with having some amount of ATOM less than the fee is still present but IMO really is just an edge case encountered during development as we're talking sub-penny amounts

<img width="526" alt="image" src="https://user-images.githubusercontent.com/17035424/193573599-67546f31-d63d-48f0-a21f-bbdacece616d.png">
<img width="526" alt="image" src="https://user-images.githubusercontent.com/17035424/193573649-c7ca589c-09bb-4b00-99f3-cae5375b9a17.png">

**Comment by willyogo** [comment]
nice! that's awesome that we've got it fixed for sending @gomesalexandre. Do we need to do anything else to make 'Max' work for staking? I just tested and am still getting "Not enough ATOM to cover gas" when attempting to stake max.

![2022-10-07 13 45 34](https://user-images.githubusercontent.com/26232238/194628259-b906068e-2151-43d0-97b1-a9fe92d62bad.gif)
